halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/KayentaProfileFactory.java

```diff
@@ -33,5 +37,55 @@ public SpinnakerArtifact getArtifact() {
   protected void setProfile(Profile profile, DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
     super.setProfile(profile, deploymentConfiguration, endpoints);
     profile.appendContents(profile.getBaseContents());
+
+    Canary canary = deploymentConfiguration.getCanary();
+
+    if (canary.isEnabled()) {
+      profile.appendContents("kayenta:");
+
+      for (AbstractCanaryServiceIntegration svc : canary.getServiceIntegrations()) {
+        if (svc instanceof GoogleCanaryServiceIntegration) {
+          GoogleCanaryServiceIntegration googleCanaryServiceIntegration = (GoogleCanaryServiceIntegration)svc;
+          boolean isGcsEnabled = googleCanaryServiceIntegration.isGcsEnabled();
+          boolean isStackdriverEnabled = googleCanaryServiceIntegration.isStackdriverEnabled();
+
+          profile
+              .appendContents("  google:")
```

Comment:
Could you use a hashmap to string implementation? There are utils for writing objects as yaml in these profile factories -- check out what gate does